src/mono/netcore/System.Private.CoreLib/src/ILLink/ILLink.Suppressions.cs

```diff
@@ -0,0 +1,201 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+
+[assembly: UnconditionalSuppressMessage ("ILLinker", "IL2001", Scope = "type", Target = "T:Mono.ValueTuple")]
```

Comment:
This seems pretty excessive to ship inside the shipping assembly if the goal is just to result in no warnings.  Does the linker provide a way that we could list granular suppressions for the purpose of our build, but then a single assembly-wide suppression in the shipping assembly?

Reply:
I would have probably assumed that the linker would actually take these into consideration for verbose warnings, but then linked them out, is that not the case?

Reply:
For assembly-wide suppression we can do this:
```C#
[assembly: UnconditionalSuppressMessage ("ILLinker", "IL2001")]
[assembly: UnconditionalSuppressMessage ("ILLinker", "IL2002")]
...
```

Reply:
But if we do assembly-wide suppressions, we won't get new warnings when writing new code in the assembly.

Reply:
> I would have probably assumed that the linker would actually take these into consideration for verbose warnings, but then linked them out, is that not the case?

My understanding is that they remain in the final binaries in the runtime-pack and shared framework, since today those are the same and need to be considered as input to customer app when linking.

> if we do assembly-wide suppressions, we won't get new warnings when writing new code in the assembly.

Can we still have granular suppressions though some mechanism when we're building?   Re-reading your original issue I see the goal is not to "always suppress" these in framework code, but only to "temporarily suppress" them until we can properly annotate everything.  I guess in that case this bloat is a point-in-time problem.

Reply:
I discussed this with @eerhardt  - there was a disconnect on our end... anyway:
* Hardcoding the suppressions into the library source code feels like a too much of a final solution (it means there's no way to "reenable" these warnings without rebuilding the library)
* We want the baselines to exist for the runtime repo build, but we don't necessarily want/need them for the application build
* Linker/SDK will have a solution to suppress all of these warnings by default anyway (https://github.com/dotnet/sdk/pull/12388) - so we don't NEED to have a solution for this in the runtime repo
* We would like to have the freedom to tweak the user experience (as in how much warnings we show to the user) later on - this means the libraries themselves should not hide anything - if we need to we will hide it in higher layers (linker or SDK)

So the technical solution we thought would be best is to:
* Encode the suppressions in linker attribute XML files (basically just like the CS, but encoded in XML - https://github.com/mono/linker/blob/master/docs/data-formats.md#custom-attributes-annotations-format)
* We will change the linker to be able to generate the suppressions in the XML format automatically - https://github.com/mono/linker/issues/1380
* We would then pass these XMLs only on the command line to the linker - so they take effect during the runtime repo build, but they don't get baked into the product.